rosbag2_py/src/rosbag2_py/format_bag_metadata.cpp

```diff
@@ -238,11 +258,27 @@ void format_service_with_type(
       info_stream << std::endl;
     };
 
-  print_service_info(services[0]);
+  std::vector<size_t> sorted_idx(services.size());
+  std::iota(sorted_idx.begin(), sorted_idx.end(), 0);
+  std::sort(
+    sorted_idx.begin(),
+    sorted_idx.end(),
+    [&services, sort_method](size_t i1, size_t i2) {
+      if (sort_method == rosbag2_py::SortingMethod::TYPE) {
+        return services[i1]->service_metadata.type < services[i2]->service_metadata.type;
+      }
+      if (sort_method == rosbag2_py::SortingMethod::COUNT) {
+        return services[i1]->event_message_count < services[i2]->event_message_count;
+      }
+      return services[i1]->service_metadata.name < services[i2]->service_metadata.name;
+    }
```

Comment:
Please use switch-case instead of if-else with returns.
In case of `default:` "unknown value" in switch case, please throw std::runtime_error("switch is not exhaustive");